boards/Kconfig

```diff
@@ -2668,6 +2675,7 @@ config ARCH_BOARD
 	default "axoloti"                  if ARCH_BOARD_AXOLOTI
 	default "bambino-200e"             if ARCH_BOARD_BAMBINO_200E
 	default "bl602evb"                 if ARCH_BOARD_BL602EVB
+	default "ch32v307evt"              if ARCH_BOARD_CH32V307EVT
```

Comment:
@robertlipe please note that board names are sorted here, so please move it to below "c5471evm"